File: backend/bike-service/src/main/java/com/bikerental/bike_service/dto/response/InternalBikeDetailsDto.java
package com.bikerental.bike_service.dto.response;

import com.bikerental.bike_service.enums.ApprovalStatus;
import com.bikerental.bike_service.enums.BikeStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class InternalBikeDetailsDto {
    private Integer bikeId;
    private BigDecimal hourlyRate;
    private BigDecimal securityDeposit;
    private Integer partnerId;
    private BikeStatus bikeStatus;
    private ApprovalStatus approvalStatus;
    private Boolean isBookable;
}

File: backend/booking-service/Infrastructure/Clients/InternalBikeDetailsDto.cs
namespace booking_service.Infrastructure.Clients
{
    public class InternalBikeDetailsDto
    {
        public int BikeId { get; set; }
        public decimal HourlyRate { get; set; }
        public decimal SecurityDeposit { get; set; }
        public int PartnerId { get; set; }
        public string BikeStatus { get; set; } = default!;
        public string ApprovalStatus { get; set; } = default!;
        public bool IsBookable { get; set; }
    }
}
